src/lib.rs

```diff
@@ -129,54 +129,54 @@ where
     }
     if let Some(build_id) = object.build_id()? {
         let path = locate_debug_build_id(build_id);
-        if path.is_ok() {
-            return path;
+        if let Some(path) = path {
+            return Ok(Some(path));
         }
         // If not found, try gnu_debuglink.
     }
     if let Some((filename, crc)) = object.gnu_debuglink()? {
         let filename = path_from_bytes(filename)?;
         return locate_gnu_debuglink(path.as_ref(), filename, crc);
     }
-    Err(anyhow!("Object does not have debug info pointer"))
+    Ok(None)
 }
 
 /// Attempt to locate the Mach-O file contained within a dSYM bundle containing the debug
 /// symbols for the Mach-O file at `path` with UUID `uuid`.
-pub fn locate_dsym<T>(path: T, uuid: Uuid) -> Result<PathBuf, Error>
+pub fn locate_dsym<T>(path: T, uuid: Uuid) -> Result<Option<PathBuf>, Error>
 where
     T: AsRef<Path>,
 {
     if let Some(dsym_path) = locate_dsym_fastpath(path.as_ref(), uuid) {
-        return Ok(dsym_path);
+        return Ok(Some(dsym_path));
     }
     locate_dsym_using_spotlight(uuid)
 }
 
 /// Attempt to locate the separate debug symbol file for the object file at `path` with
 /// build ID `id`.
-pub fn locate_debug_build_id(id: &[u8]) -> Result<PathBuf, Error> {
+pub fn locate_debug_build_id(id: &[u8]) -> Option<PathBuf> {
```

Comment:
```suggestion
pub fn locate_debug_build_id(id: &[u8; 20]) -> Option<PathBuf> {
```

Reply:
From our conversation in this PR, we want to keep this a slice.